strawberryfields/parameters.py

```diff
@@ -1,64 +1,78 @@
+# limitations under the License.
 """
-Gate parameters
-===============
+Quantum operation parameters
+============================
 
 **Module name:** :mod:`strawberryfields.parameters`
 
 .. currentmodule:: strawberryfields.parameters
 
-The :class:`Parameter` class is an abstraction of a parameter passed to the
-quantum circuit operations represented by :class:`~strawberryfields.ops.ParOperation`.
-The parameter objects can represent a number, a NumPy array, a value measured from the quantum register
-(:class:`~strawberryfields.engine.RegRefTransform`),
-or a TensorFlow object.
+The :class:`Parameter` class encapsulates a parameter passed to the
+quantum operations represented by :class:`.Operation`.
+The parameter objects can represent a
 
-.. currentmodule:: strawberryfields
+* number
+* NumPy array
+* value measured from the quantum register (:class:`.RegRefTransform`)
+* TensorFlow object
 
-The normal lifecycle of a ParOperation object and its associated Parameter instances is as follows:
 
-* A ParOperation instance is created, and given some parameters as input.
+The normal lifecycle of an Operation object and its associated Parameter instances is as follows:
 
-* :meth:`ParOperation.__init__` converts the inputs into Parameter instances.
-  Plain :class:`~engine.RegRef` instances are wrapped in a trivial
-  :class:`~engine.RegRefTransform`.
+* An Operation instance is constructed, and given some input arguments.
+
+* :meth:`.Operation.__init__` converts the inputs into Parameter instances.
+  Plain :class:`.RegRef` instances are wrapped in a trivial :class:`.RegRefTransform`.
   RegRefTransforms add their RegRef dependencies to the Parameter and consequently to the Operation.
 
-* The Operation instance is applied using its :func:`~ops.Operation.__or__`
-  method inside an :class:`~engine.Engine` context.
-  This creates a :class:`~engine.Command` instance that wraps
-  the Operation and the RegRefs it acts on, which is appended to the Engine command queue.
+* The Operation instance is applied using its :meth:`~ops.Operation.__or__`
+  method inside an :class:`.Program` context.
+  This creates a :class:`.Command` instance that wraps
+  the Operation and the RegRefs it acts on, which is appended to the Program command queue.
 
-* Once the entire program is inputted, Engine optimizes it. This involves merging and commuting Commands
-  inside the circuit graph, the building of which requires knowledge of their dependencies, both direct and Parameter-based.
+* Once the entire program is inputted, it is compiled and optimized. This involves
+  merging and commuting Commands inside the circuit graph. The circuit graph is built
+  using the knowledge of which subsystems the Commands act and depend on.
 
-* Merging two :class:`~ops.Gate` instances of the same subclass involves
+* Merging two :class:`.Gate` instances of the same subclass involves
   adding their first parameters after equality-comparing the others. This is easily done if
```

Comment:
This is not strictly true - gates which accept matrix parameters require matrix multiplication, and some require addition of more than one parameter (`GaussianState`). We should mark this an issue to fix/implement.

Reply:
Ah, but those are not `Gate` subclasses, but rather `Decomposition` subclasses. `Gate` subclasses are defined to behave as one-parameter unitary groups wrt. p[0].

Reply:
ah, got it